### PR TITLE
First draft of documentation lesson

### PR DESCRIPTION
**Pull request by mdippery**
First draft of lesson. This one was a bit tough; I didn't want to delve too deeply into methods and objects yet, but it does touch on those concepts a bit. I also didn't think it was a good time to introduce operators like `+`, even for string concatenation, but if you think we should cover that in this lesson, I'll be happy to add it in. There might be some other things I missed, too. I think the lesson flows well enough, but I wasn't sure what was appropriate to talk about just yet. Might need some work on this, but feedback would be good at this point.

Kudos for using Audrey Hepburn, though, as she's one of my favorite actresses!

cc: @AnnJohn @JimCampagno

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,65 +1,86 @@
...
+Let's go back to your playground file and see this help documentation in action. Here's your goal: Capitalize your string "Audrey" so it becomes "AUDREY". Impossible? No way! Let's take this step-by-step so you can learn how to do this with the help of Xcode.
+
+On a new line of your playground, type `name.`—make sure you type the period at the end! Then stop. You will see a popup with a list of methods that are available to `String`s appear like this:
+
+![String autocomplete](.images/autocomplete.png)
+
+This popup is referred to as "autocomplete". This menu appears in Xcode whenever you try to call a method on a constant or variable.
+
+Remember, a _method_ is just a function that is attached to a particular data type. Methods are called like functions, except you put the constant or variable you are call on them _before_ the method name, followed by a dot, and then the name of the method. Arguments to the method are supplied the same way they are when you call functions: By placing them in parentheses after the method name. For now, just think of a method as a function. You'll learn a lot more about methods in a later lesson.
+
```
Comment (JimCampagno) [inline README.md:63]: This is a great explanation.

File: README.md [hunk 2]
```diff
@@ -1,65 +1,86 @@
...
+Swift's built-in data types come with a set of behaviors, known as _methods_, that let you use and manipulate those data types in various common ways. Both `String` and `Int` have a set of methods that let you do things to those values. Of course, you have to be able to know what methods Swift provides out of the box! That's where Xcode's help functionality comes into play.
+
+### Back to Xcode
+
+Let's go back to your playground file and see this help documentation in action. Here's your goal: Capitalize your string "Audrey" so it becomes "AUDREY". Impossible? No way! Let's take this step-by-step so you can learn how to do this with the help of Xcode.
+
+On a new line of your playground, type `name.`—make sure you type the period at the end! Then stop. You will see a popup with a list of methods that are available to `String`s appear like this:
+
+![String autocomplete](.images/autocomplete.png)
+
```
Comment (JimCampagno) [inline README.md:59]: This is exactly what I had in mind.

**Comment by JimCampagno** [comment]
> I also didn't think it was a good time to introduce operators like +, even for string concatenation, but if you think we should cover that in this lesson, I'll be happy to add it in.  

I'm fine with it not being here. I was questioning it as well.

> There might be some other things I missed, too.

No, you covered everything I had in mind. I'm thinking a little more about this lesson, maybe we also show them `lowercaseString` as well? Or have them create a new variable using `uppercaseString` like so:

``` swift
let name = "Audrey"
let uppercaseName = name.uppercaseString

print("\(name)... \(name)... \(uppercaseName)... WAKE UP!")
// prints "Audrey... Audrey... AUDREY... WAKE UP!"
```

> Might need some work on this, but feedback would be good at this point.

This was a tough lesson and I think you did a really good job. The only thing I'm thinking about adding is the whole wake up Audrey thing I wrote. But do you think that it might be too confusing to create a new constant `uppercaseName` to store this?

@mdippery

**Comment by mdippery** [comment]
@JimCampagno, that example looks good. I went ahead and added it.

**Comment by JimCampagno** [comment]
@mdippery 👍

**Pull request merged**